src/compiler.ts

```diff
@@ -6992,6 +6992,8 @@ export class Compiler extends DiagnosticEmitter {
       if (classInstances) {
         for (let _values = Map_values(classInstances), j = 0, l = _values.length; j < l; ++j) {
           let classInstance = _values[j];
+          // Chcek if the parent class is a subtype of instance's class
+          if (!classInstance.isAssignableTo(<Class>instance.parent)) continue;
```

Comment:
The cast here can be moved up to the beginning of the function so we don't need to property-access-and-cast each time in the loop :)

Reply:
Also should add an assert before the cast to make sure. We are typically doing that in other places where we rely on `kind` to have a particular value.

Reply:
Not sure where I would put it since I need access to the current `_values[j]`.

Reply:
`instance` is passed into the function as an argument, so that can be moved up:

```ts
var parent = instance.parent;
assert(parent.kind == ElementKind.CLASS);
var parentClassInstance = <Class>parent;
```

To then here become:

```ts
if (!classInstance.isAssignableTo(parentClassInstance ) continue;
```

Reply:
Ah got it.